pkg/virt-controller/watch/vm.go

```diff
@@ -479,6 +479,15 @@ func (c *VMController) handleDataVolumes(vm *virtv1.VirtualMachine, dataVolumes
 			}
 		}
 		if !exists {
+			// Don't create DV if PVC already exists
+			pvc, err := c.getPersistentVolumeClaimFromCache(vm.Namespace, template.Name)
+			if err == nil && pvc != nil {
+				populatedFor := pvc.Annotations["cdi.kubevirt.io/storage.populatedFor"]
```

Comment:
Is this annotation added when DV is GC'd?  Otherwise, it shouldn't always be there

Reply:
No, it's already used in the restore for creating a DV when its populated PVC already exists.
@ShellyKa13 ?

Reply:
yes in the restore in case of datavolumetemplate we update the populatedFor to be the name of the new dvrestore name

Reply:
Not sure what is going on here is correct.  I don't think "populatedFor" should have any effect here.  It is used for more things than vm restore, btw.

As-is, it appears to me that any "populatedFor" DataVolumes will never be garbage collected.  Or more precisely, they will always get recreated after they are garbage collected.

I think VM controller should simply only create the DV if the PVC does not exist.

That means that the restore controller has to create the DVs during restore, though.

Reply:
> As-is, it appears to me that any "populatedFor" DataVolumes will never be garbage collected. Or more precisely, they will always get recreated after they are garbage collected.

not sure I understand why they will always be recreated, actually this code exactly should make sure that they will not be recreated (if annotation exists continue).
But either way I do understand what you are saying regarding the datavolume should be created in the restore and I agree, I did wonder why we create the pvc but not the datavolume in the restore process.

Reply:
Agree. I guess the `populatedFor` check is a leftover. Removing it, so no DV will be created in the restore.

Reply:
@ShellyKa13 it only continued if `populatedFor != template.Name` which will not the case with restore

Reply:
> I did wonder why we create the pvc but not the datavolume in the restore process.

Eh basically less code, makes restore simpler if we can depend on some preexisting code path

> The question is if we decided that the default behavior is as if gc is on then on the case of restore there is no need to create the data volumes at all since the pvcs are already created from the volumesnapshots.

Yeah there is no "need" to create the datavolumes but if they don't get created that is a significant change in behavior when gc is disabled.

Reply:
Do we really need to be afraid of this `significant change in behavior when gc is disabled` which actually simplifies code?

Reply:
> @ShellyKa13 it only continued if `populatedFor != template.Name` which will not the case with restore

why in the restore we update the template name to be the name of the restored pvc

Reply:
> Yeah there is no "need" to create the datavolumes but if they don't get created that is a significant change in behavior when gc is disabled.

right. in case we dont have gc then to not create the datavolume is a real change in behavior. 


> which actually simplifies code?

It simplifies code for the gc for sure.. although for gc it wont change much in terms that the restore will create the datavolume and just gc it later, comparing to not having gc and just changing the behavior in restore to not create the datavolume at all not sure that makes sense without having gc enabled..
to conclude if we do want to create the datavolume I agree it makes more sense to move that logic to the restore controller. The question is if we want to. and I don't have a strong opinion either way. I think the correct way is to create it in the restore and it will be GCed later if GC enabled or not, but Im also ok with not creating it at all..

Reply:
I think the "right" thing to do is to have restore create the DV even if it is garbage collected right away.  BUT this may be a tricky thing to do if GC is enabled.  How do we track that a DataVolume was created successfully if it may be garbage collected right away?  May have to persist in restore status.  Basically we have to make sure that the restore code does not consistently keep creating DVs.

Reply:
@mhenriks things get a bit complex with your "right" direction. I don't see why not keep restore consistent with the new behavior, so if PVC exists DV is not created. Afaik no KubeVirt code currently requires/refers the DV in the restore flow, and all CI lanes look green with this behavior for both GC enabled/disabled.

Reply:
I really don't like the idea of mixing this compact PR with improvement of the restore DV creation. We sure won't duplicate the VM controller logic, so a quite heavy refactoring will be needed there.
However, I understand that for some reason (preparation for DV source = volume snapshot?) you don't want to get rid of the restore DV creation when PVC exist. 
Let's consider this simple solution: when restore creates the PVC it is annotated with `dvRequired`, so when VM controller encounters PVC with that annotation, it creates the DV and removes the PVC annotation.
wdyt?

Reply:
I just thought of an easy way to track whether restore controller created the DataVolume.  Initially, the restore PVC has no owner.  When DV is created, it will take ownership of the PVC.  When the DV is garbage collected, the VM will own the PVC.  So, restore process will only create the DV when it does not exist and the PVC has no owner.  What do you think?

As you can probably guess, my preference is to not change existing behavior unless there is a very good reason.

Reply:
Looks like a clean logic, so why not keep DV creation in the VM controller?